src/Themes.vala

```diff
@@ -40,12 +40,34 @@ public class Terminal.Themes {
 
     // format is color01:color02:...:color16:background:foreground:cursor
     public static Gdk.RGBA[] get_rgba_palette (string theme) {
-        const string[] DARK_PALETTE = {
-            "#073642", "#dc322f", "#859900", "#b58900", "#268bd2", "#ec0048", "#2aa198", "#94a3a5",
-            "#586e75", "#cb4b16", "#859900", "#b58900", "#268bd2", "#d33682", "#2aa198", "#6c71c4",
-            "rgba(46, 46, 46, 0.95)", "#a5a5a5", "#839496"
-        };
+        var string_palette = get_string_palette (theme);
+        bool settings_valid = string_palette.length == PALETTE_SIZE;
+
+        var rgba_palette = new Gdk.RGBA[PALETTE_SIZE];
+        for (int i = 0; i < PALETTE_SIZE; i++) {
+            var new_color = Gdk.RGBA ();
+            // Replace invalid color with corresponding one from default palette
+            if (!new_color.parse (string_palette[i])) {
+                critical ("Color %i '%s' is not valid - replacing with default", i, string_palette[i]);
+                string_palette[i] = DARK_PALETTE[i];
+                new_color.parse (DARK_PALETTE[i]);
+                settings_valid = false;
+            }
+
+            rgba_palette[i] = new_color;
+        }
 
+        return rgba_palette;
+    }
+
+    // DARK_PALETTE is used as a fallback when custom palette colors are invalid
+    public const string[] DARK_PALETTE = {
```

Comment:
Since we have `get_string_palette` now, should this not be a constant and have `get_rgba_palette` use `get_string_palette (prefer_dark ? DARK : LIGHT)` for its fallback?

Reply:
This is possible only if we replace the *whole* of the palette if *any* of the custom colors is invalid.  Is that OK? At the moment, we only replace individual colors.